tools/circt-lec/circt-lec.cpp

```diff
@@ -122,6 +122,65 @@ static cl::opt<OutputFormat> outputFormat(
 // Tool implementation
 //===----------------------------------------------------------------------===//
 
+// Move all operations in `src` to `dest`. Rename all symbols in `src` to avoid
+// conflict.
+static FailureOr<StringAttr> mergeModules(ModuleOp dest, ModuleOp src,
+                                          StringAttr name) {
+
+  SymbolTable destTable(dest), srcTable(src);
+  StringAttr renamed = {};
+  for (auto &op : src.getOps()) {
+    if (SymbolOpInterface symbol = dyn_cast<SymbolOpInterface>(op)) {
+      auto oldSymbol = symbol.getNameAttr();
+      auto result = srcTable.renameToUnique(&op, {&destTable});
+      if (failed(result))
+        return src->emitError() << "failed to rename symbol " << oldSymbol;
+
+      if (oldSymbol == name) {
+        assert(!renamed && "symbol must be unique");
+        renamed = *result;
+      }
+    }
+  }
+
+  if (!name)
+    return src->emitError()
+           << "module " << name << " was not found in the second module";
```

Comment:
I'm a little confused on this error message - it looks like it's throwing when a name isn't given, rather than when a name isn't found? Maybe I missed some detail though!

Reply:
Good catch, the condition must be `!newName`. I added a test :)